tests/unit-tests/test_nn_model.py

```diff
@@ -224,6 +228,295 @@ def test_hgt_with_no_indegree_dstnodes(input_dim, output_dim):
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb2 = layer(block, node_feats)
+    # check output numbers, and dimensions
+    assert 'n0' not in emb2
+    assert emb2['n1'].shape[0] == len(seeds['n1'])
+    assert emb2['n1'].shape[1] == output_dim
+
+    # Test case 3: normal case, all 5 message passong ops
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        edge_feat_mp_op="add",
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    edge_feats = {
+        ("n0", "r0", "n1"): th.rand(r0_eid.shape[0], input_dim).to(dev),
+        ("n0", "r1", "n1"): th.rand(r1_eid.shape[0], input_dim).to(dev)
+    }
+
+    emb3 = layer(block, node_feats, edge_feats)
+    # check output numbers, dimensions and device
+    assert 'n0' not in emb3
+    assert emb3['n1'].shape[0] == len(seeds['n1'])
+    assert emb3['n1'].shape[1] == output_dim
+
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        edge_feat_mp_op="sub",
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb3 = layer(block, node_feats, edge_feats)
+    # check output numbers, dimensions and device
+    assert 'n0' not in emb3
+    assert emb3['n1'].shape[0] == len(seeds['n1'])
+    assert emb3['n1'].shape[1] == output_dim
+
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        edge_feat_mp_op="mul",
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb3 = layer(block, node_feats, edge_feats)
+    # check output numbers, dimensions and device
+    assert 'n0' not in emb3
+    assert emb3['n1'].shape[0] == len(seeds['n1'])
+    assert emb3['n1'].shape[1] == output_dim
+
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        edge_feat_mp_op="div",
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb3 = layer(block, node_feats, edge_feats)
+    # check output numbers, dimensions and device
+    assert 'n0' not in emb3
+    assert emb3['n1'].shape[0] == len(seeds['n1'])
+    assert emb3['n1'].shape[1] == output_dim
+
+    # Test case 4: abnormal case, layer has no edge feature weights, but give edge features
+    #              the edge features will be ignor. layer only use node features in forward()
```

Comment:
How to test the edge features are ignored?

Reply:
Shall we check the output of layer(block, node_feats) is same as layer(block, node_feats, edge_feats)?

Reply:
Added a test.